### PR TITLE
osc 0.154.0 (new formula)

### PR DESCRIPTION
**Pull request by catap**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [X] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

---

### CODE REVIEW

**Comment by DomT4** [comment]
`pycurl` and `urlgrabber` should be vendored into the `osc` formula as `resource`s rather than adding unique formulae for them. We don't accept Python _libraries_ in the core, barring exceptional circumstances. Example [here](https://github.com/Homebrew/homebrew-core/blob/fb81645562f7e495a49561065912788af2d1f14f/Formula/mkdocs.rb#L76-L99).

**Comment by catap** [comment]
@DomT4 thanks! Fixed.

File: Formula/osc.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
+class Osc < Formula
+  desc "The Command Line Interface to work with an Open Build Service"
+  homepage "https://github.com/openSUSE/osc"
+  url "https://github.com/openSUSE/osc/archive/0.154.0.tar.gz"
+  sha256 "88daae5b94354e9bcab03523aa7e3d270f69ffeaef8e4a1493bce19757d4699d"
+  head "https://github.com/openSUSE/osc.git"
+
+  bottle :unneeded
```
Comment (DomT4) [inline Formula/osc.rb]: You do need a bottle here. Scrap this line.

File: Formula/osc.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    sha256 "06b13ff8d527dba3aee04069681b2c09c03117592d5485a80ae4b807cdf33476"
+  end
+
+  resource "m2crypto" do
+    url "https://pypi.python.org/packages/packages/source/M/M2Crypto/M2Crypto-0.24.0.tar.gz"
+    sha256 "80a56441a1d2c0cf27e725be7554c92598b938fc8767ee2c71fdbc2fdc055ee8"
+  end
+
+  # Fix path to openssl default certificate
+  patch :DATA
```
Comment (DomT4) [inline Formula/osc.rb]: Is this something we can report upstream & ask them to make a little more flexible?

Comment (DomT4) [inline Formula/osc.rb]: We can use an `inreplace` inside `def install` at least, which is neater:

```ruby
inreplace "osc/conf.py", "'/etc/ssl/certs'", "'#{etc}/openssl/cert.pem'"
```

**Comment by tdsmith** [comment]
I might nick the comment on line 31 but otherwise this lgtm!

**Comment by DomT4** [comment]
Killed it in post.

Merged in https://github.com/Homebrew/homebrew-core/commit/b76f044627f30d881bc47447de4e2d930186c5f3. Thank you for another contribution to Homebrew @catap; we appreciate it! :smiley_cat:

**Pull request closed without merging**